Forta-Bots-Challenge-2/src/agent.spec.ts

```diff
@@ -0,0 +1,216 @@
+import { HandleTransaction, Initialize } from "forta-agent";
+import { provideHandleTransaction, provideInitialize } from "./agent";
+import { UNISWAP_POOL_ABI, ERC20_ABI, SWAP_EVENT } from "./constants";
+import { MockEthersProvider, TestTransactionEvent } from "forta-agent-tools/lib/test";
+import { createAddress } from "forta-agent-tools";
+import { Interface, defaultAbiCoder, getCreate2Address, solidityKeccak256, keccak256 } from "ethers/lib/utils";
+import { createTestFinding } from "./findings";
+import { BigNumber } from "ethers";
+import { PoolData } from "./uniswap.fetcher";
+
+// format: [blockNumber, token0, token1, fee]
+const TEST_DATA: [number, string, string, BigNumber][] = [
+  [10, createAddress("0xaa1"), createAddress("0xbb1"), BigNumber.from(1)],
+  [20, createAddress("0xaa2"), createAddress("0xbb2"), BigNumber.from(2)],
+  [30, createAddress("0xaa3"), createAddress("0xbb3"), BigNumber.from(3)],
+];
+
+// format: [sender,  recipient,  amount0, amount1, sqrtPriceX96, liquidity, tick]
+const TEST_LOGS: [string, string, BigNumber, BigNumber, BigNumber, BigNumber, BigNumber][] = [
+  [
+    createAddress("0xae1"),
+    createAddress("0xbe2"),
+    BigNumber.from(10),
+    BigNumber.from(10),
+    BigNumber.from(1),
+    BigNumber.from(100),
+    BigNumber.from(1),
+  ],
+  [
+    createAddress("0xae2"),
+    createAddress("0xce3"),
+    BigNumber.from(5),
+    BigNumber.from(15),
+    BigNumber.from(2),
+    BigNumber.from(50),
+    BigNumber.from(2),
+  ],
+  [
+    createAddress("0xab6"),
+    createAddress("0xcb5"),
+    BigNumber.from(50),
+    BigNumber.from(1),
+    BigNumber.from(20),
+    BigNumber.from(500),
+    BigNumber.from(3),
+  ],
+];
+
+const TEST_POOL_DATA_0: PoolData = {
```

Comment:
Similar to the above test variables, these could also be in an array of type `PoolData[]`.